### PR TITLE
try to detect the type of inserted field before changing it to json

### PR DESCRIPTION
**Pull request by MarketaAdamova**

### CODE REVIEW

File: src/utility_belt/sql/conv.clj [hunk 1]
```diff
@@ -60,6 +60,11 @@
...
   IPersistentVector
-  (set-parameter [value statement idx]
-    (.setObject ^PreparedStatement statement idx
-                (value-to-json-pgobject value))))
+  (set-parameter [v ^PreparedStatement s ^long i]
+   (let [conn (.getConnection s)
+          meta (.getParameterMetaData s)
+          type-name (.getParameterTypeName meta i)]
+     (if-let [elem-type (when type-name (second (re-find #"^_(.*)" type-name)))]
+       (.setObject s i (.createArrayOf conn elem-type (to-array v)))
```
Comment (lukaszkorecki) [inline src/utility_belt/sql/conv.clj:68]: Check this out: https://github.com/seancorfield/next-jdbc/blob/develop/src/next/jdbc/types.clj - next.jdbc has a shortcut ns for turning things into types

Comment (MarketaAdamova) [reply src/utility_belt/sql/conv.clj:68]: @lukaszkorecki  I tried to use it but it doesn't solve the issue I was having (well I can't get the tests to pass in lucien using that ns). Sometimes we want the things to be turned into json and sometimes we want vector of uuids etc.

Comment (lukaszkorecki) [reply src/utility_belt/sql/conv.clj:68]: Right, moving on then.

**Review by lukaszkorecki — APPROVED** [review]
I’d verify we don’t get any reflection warnings 👍

**Pull request merged**